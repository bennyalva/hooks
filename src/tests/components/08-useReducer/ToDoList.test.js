import React from 'react';
const { shallow } = require("enzyme");
const { ToDoList } = require("../../../components/08-useReducer/ToDoList");
const { demoToDos } = require("../../fixtures/demoToDos");

describe('Tests ToDoList', () => {
    const handlerDelete = jest.fn();
    const handlerToggle = jest.fn();

    const wrapper = shallow(<ToDoList 
        toDos={demoToDos}
        handlerDelete={handlerDelete}
        handlerToggle={handlerToggle}
       />)
   test('should show correctly', () => {
      expect(wrapper).toMatchSnapshot(); 
   });
   test('should get two <TodoListItem />', () => {
    expect(wrapper.find('ToDoListItem').length).toBe(demoToDos.length);
    expect(wrapper.find('ToDoListItem').at(0).prop('handlerDelete')).toEqual(expect.any(Function))
    expect(wrapper.find('ToDoListItem').at(0).prop('handlerToggle')).toEqual(expect.any(Function))
    
 });
    
});
