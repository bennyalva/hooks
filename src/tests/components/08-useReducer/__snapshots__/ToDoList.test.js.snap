// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests ToDoList should show correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <ToDoListItem
    handlerDelete={[MockFunction]}
    handlerToggle={[MockFunction]}
    i={0}
    key="1"
    toDO={
      Object {
        "desc": "react",
        "done": false,
        "id": 1,
      }
    }
  />
  <ToDoListItem
    handlerDelete={[MockFunction]}
    handlerToggle={[MockFunction]}
    i={1}
    key="2"
    toDO={
      Object {
        "desc": "mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
